### PR TITLE
[CM-1955] Expose Dynamic online/offline status update method

### PR DESCRIPTION
**Pull request by AbhijeetRanjan308**
## Summary
- Exposed the code for Update User Status in Android and iOS

## Code
```
 final dynamic updateStatusObject = {
         'assigneID': "<Assignee-ID>",
         'status': "away",
    };

 KommunicateFlutterPlugin.updateUserStatus(updateStatusObject);
```

## Parameters
- assigneID : "" or "Assignee-ID". (compulsory to pass this as well either empty or with value)
- status : away, online, offline, default (capital character's are also accepted)

### CODE REVIEW

File: android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java [hunk 1]
```diff
@@ -105,6 +106,55 @@ public void onFailure(RegistrationResponse registrationResponse, Exception excep
             } catch (Exception e) {
                 result.error(ERROR, e.getMessage(), null);
             }
+        } else if (call.method.equals("updateStatus")) {
+            try {
+                JSONObject updateStatusObjc = new JSONObject(call.arguments.toString());
+                if (updateStatusObjc.has("assigneID") && updateStatusObjc.has("status")) {
+                    final String AssigneeID = updateStatusObjc.get("assigneID").toString();
+                    final String Status = updateStatusObjc.get("status").toString();
```
Comment (Sathyan-KM) [inline android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java:114]: check final naming convention

File: android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java [hunk 2]
```diff
@@ -105,6 +106,55 @@ public void onFailure(RegistrationResponse registrationResponse, Exception excep
...
+                                Kommunicate.updateAssigneeStatus(AssigneeID, KMAgentStatusHelper.KMAgentStatus.OFFLINE);
+                                break;
+                            case "away":
+                                Kommunicate.updateAssigneeStatus(AssigneeID, KMAgentStatusHelper.KMAgentStatus.AWAY);
+                                break;
+                            default:
+                                Kommunicate.updateAssigneeStatus(AssigneeID, KMAgentStatusHelper.KMAgentStatus.DefaultStatus);
+                                break;
+                        }
+                    }
```
Comment (Sathyan-KM) [inline android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java:150]: Can we use single switch case without if  & else  & a variable to handle the nullability of assignee ID

File: ios/Classes/SwiftKommunicateFlutterPlugin.swift [hunk 3]
```diff
@@ -131,6 +131,38 @@ public class SwiftKommunicateFlutterPlugin: NSObject, FlutterPlugin, KMPreChatFo
...
             }
             Kommunicate.updatePrefilledText(prefilledText)
+        } else if(call.method == "updateStatus") {
+            guard let jsonString = call.arguments as? String, var jsonObj = jsonString.convertToDictionary(), let assigneeID = jsonObj["assigneID"] as? String, let status = jsonObj["status"] as? String else {
+                self.sendErrorResultWithCallback(result: result, message: "Unable to parse update user status Object")
+                return
+            }
+            if status.isEmpty {
+                self.sendErrorResult(message: "Status is Empty.")
+                return
```
Comment (Sathyan-KM) [inline ios/Classes/SwiftKommunicateFlutterPlugin.swift:141]: use another guard let for status empty check & variable for assigneeid . And then use it in a single switch

**Review by Sathyan-KM — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java (hunk 2 above)
Comment (AbhijeetRanjan308) [reply android/src/main/java/io/kommunicate/kommunicate_flutter_plugin/KmMethodHandler.java:150]: okay

File: ios/Classes/SwiftKommunicateFlutterPlugin.swift (hunk 3 above)
Comment (AbhijeetRanjan308) [reply ios/Classes/SwiftKommunicateFlutterPlugin.swift:141]: okay

Comment (AbhijeetRanjan308) [reply ios/Classes/SwiftKommunicateFlutterPlugin.swift:141]: Not required already have a check above this

**Review by Sathyan-KM — APPROVED** [review]

**Review by Sathyan-KM — APPROVED** [review]

**Pull request merged**